src/olympia/files/tests/test_helpers.py

```diff
@@ -70,6 +70,16 @@ def test_recurse_contents(self):
         for name in nm:
             assert name in files
 
+    def test_recurse_contents_of_zip(self):
+        self.viewer.src = get_file('recurse.zip')
+        self.viewer.extract()
+        files = self.viewer.get_files()
+        nm = ['recurse/recurse.xpi/chrome/test-root.txt',
+              'recurse/somejar.jar/recurse/recurse.xpi/chrome/test.jar',
+              'recurse/somejar.jar/recurse/recurse.xpi/chrome/test.jar/test']
+        for name in nm:
+            assert name in files
```

Comment:
I'm usually not a fan of testing things in `for` statements since it could mask or hide things, especially not if it you're actually introducing more code by doing so :) BUT in this example it's more code-style so I'm torn but more on the "yeah, I like this better" side of things...

so, uhm, I'm just speaking with myself here :laughing: